wlauto/workloads/adobereader/uiauto/app/src/main/java/com/arm/wlauto/uiauto/adobereader/UiAutomation.java

```diff
@@ -102,6 +102,10 @@ private void dismissWelcomeView() throws Exception {
         if (!onboarding_finish_button.exists()) {
             welcomeView.swipeLeft(10);
         }
+        
+        if (!onboarding_finish_button.exists()) {
+        return;
+        }
```

Comment:
From what I can tell the additional actions are for after the on-boarding button has been pressed which would be upon initial loading of the main screen and therefore may still be applicable. 
In particular the last check of the function is to ensure that the main screen has finished loading and is ready to continue with the workload which without, could cause issues on slower devices.